components/color-picker/index.jsx

```diff
@@ -0,0 +1,364 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+import classNames from 'classnames';
+
+import checkProps from './check-props';
+
+import CustomColor from './private/custom-color';
+import Swatch from './private/swatch';
+import SwatchPicker from './private/swatch-picker';
+
+import Button from '../button';
+import Dialog from '../utilities/dialog';
+import Input from '../forms/input';
+import Tabs from '../tabs';
+import TabsPanel from '../tabs/panel';
+
+import ColorUtils from '../../utilities/color';
+
+import { COLOR_PICKER } from '../../utilities/constants';
+
+const propTypes = {
+	/**
+	 * **Assistive text for accessibility**
+	 * * `hueSlider`: Instructions for hue selection input
+	 * * `saturationValueGrid`: Instructions for using the grid for saturation
+	 * and value selection
+	 */
+	assistiveText: PropTypes.shape({
+		hueSlider: PropTypes.string,
+		saturationValueGrid: PropTypes.string,
+	}),
+	/**
+	 * Current color (in hexadecimal string)
+	 */
+	value: PropTypes.string,
+	/**
+	 * Unique ID for component
+	 */
+	id: PropTypes.string,
+	/**
+	 * Disables the input and button
+	 */
+	disabled: PropTypes.bool,
+	/**
+	 * Popover open state
+	 */
+	isOpen: PropTypes.bool,
+	/**
+	 * **Text labels for internationalization**
+	 * * `blueAbbreviated`: One letter abbreviation of blue color component
+	 * * `cancelButton`: Text for cancel button on popover
+	 * * `customTab`: Text for custom tab of popover
+	 * * `greenAbbreviated`: One letter abbreviation of green color component
+	 * * `hexLabel`: Label for input of hexadecimal color
+	 * * `invalidColor`: Error message when hex color input is invalid
+	 * * `invalidComponent`: Error message when a component input is invalid
+	 * * `label`: Label for entire colorpicker control
+	 * * `redAbbreviated`: One letter abbreviation of red color component
+	 * * `swatchTab`: Label for swatch tab of popover
+	 * * `submitButton`: Text for submit/done button of popover
+	 */
+	labels: PropTypes.shape({
+		blueAbbreviated: PropTypes.string,
+		cancelButton: PropTypes.string,
+		customTab: PropTypes.string,
+		greenAbbreviated: PropTypes.string,
+		hexLabel: PropTypes.string,
+		invalidColor: PropTypes.string,
+		invalidComponent: PropTypes.string,
+		label: PropTypes.string,
+		redAbbreviated: PropTypes.string,
+		swatchTab: PropTypes.string,
+		submitButton: PropTypes.string
+	}),
+	/**
+	 * Triggered when the date changes. It receives and event object that
+	 * originally triggered the change, as well as an object in the shape `{
+	 * color: [string] }`, which is a hex representation of the color.
+	 */
+	onChange: PropTypes.func,
+	/**
+	 * An array of hex color values which is used to set the options of the
+	 * swatch tab of the colorpicker popover.
+	 */
+	swatchColors: PropTypes.arrayOf(PropTypes.string),
+};
```

Comment:
We need to add a way to open the dialog (we do with `isOpen`) and select a tab. `props.selector` with `swatches` and `custom` as options. Panel selection would only work with `base` variant. This would make the tab/panel selection _controlled_ and the Colorpicker would need to fire an event off when the other tab was clicked. Something like `onSelectorChange` with the data parameter of `swatches` and `custom`.

We need a variant prop with `custom`, `base`, `predefined` and `swatches` as options that removes the tab UI.